bonsai/ethereum-relay/README.md

```diff
@@ -9,77 +9,118 @@ The picture below shows a simplified overview of how users can integrate Bonsai
-4. Both proof and journal are sent on-chain to the `Bonsai Relay Contract`, that verifies the correctness of the proof.
-5. Upon successful verification, the journal gets sent to the user contract via the specified callback function.
+1. Users can delegate their smart contract's logic to Bonsai. The `Bonsai Relay` provides a `Request Callback` interface. This interface, accessible both *off-chain* (through HTTP REST API) and *on-chain*, emits an event detected by the `Ethereum Bonsai Relayer`.
+2. The `Ethereum Bonsai Relayer` sends the proof request to Bonsai.
+3. Bonsai generates a Snark proof and its result, encapsulated in a journal.
+4. The `Ethereum Bonsai Relayer` submits this proof and journal on-chain to the `Bonsai Relay Contract` for validation.
+5. If validated, the journal is dispatched to the user's smart contract via the specified callback.
 
 ## Interfaces
 
-The Bonsai Ethereum Relay provides both an *on-chain* (via Ethereum) and an *off-chain* (via HTTP REST API) interface to send `Callback requests`.
+The Bonsai Ethereum Relay provides both an *off-chain* (via HTTP REST API) and an *on-chain* (via Ethereum) interface to send `Callback requests`.
 
-### On-chain
+### Off-chain
 
 A typical flow works as follows:
 
-1. Deploy a Bonsai Relay Smart Contract on Ethereum at a given address `0xB..`.
-2. Start an instance of the relay tool configured with the option `--contract-address` defined as `0xB..`.
-3. Delegate some off-chain computation for a given Smart Contract `A` to Bonsai by registering the `Image` or `ELF` (i.e., the compiled binary responsible for executing the given computation on the RISC Zero ZKVM) to Bonsai.
-4. The corresponding `Image ID` and the Bonsai Relay Smart Contract `0xB..` can be used to construct and deploy the Smart Contract `A` to Ethereum.
-5. Send a transaction to Smart Contract `A` to trigger a `Callback request` event that the Bonsai Relay will catch and forward to Bonsai.
-6. Once Bonsai has generated a proof of execution, the Bonsai Relay will forward this proof along with the result of the computation to the Bonsai Relay Smart Contract.
-7. This triggers a verification of the proof on-chain, and only upon successful verification, the result of the computation will be forwarded to the original requester Smart Contract `A` via the `invoke_callback` function.
-
-### REST API
-
-As an alternative to sending a `Callback request` from Ethereum as described by step 5, the request can be sent directly to the Bonsai Relay via an HTTP REST API.
-Then, the remaining steps will flow as above. The following example explains how to do that.
+1. Deploy the `Bonsai Relay Contract` to Ethereum using address `0xB..`.
+2. Launch the `Bonsai Relayer`, setting `--contract-address` to `0xB..`.
+3. To delegate off-chain computation for Smart Contract `A` to Bonsai, register its `Image` or `ELF` (the binary executing the computation on RISC Zero ZKVM) with Bonsai.
```

Comment:
I was a bit confused about whether "Smart Contract `A`" was a contract mentioned before that I was supposed to know something about. Maybe this is clearer?
```suggestion
3. To delegate off-chain computation for a new Smart Contract `A` to Bonsai, register its `Image` or `ELF` (the binary executing the computation on RISC Zero ZKVM) with Bonsai.
```